QStore4S/src/main/java/edu/asu/qstore4s/async/manager/impl/AsyncQueryManager.java

```diff
@@ -0,0 +1,116 @@
+ *
+ */
+@Service
+public class AsyncQueryManager implements IAsyncQueryManager {
+
+    @Autowired
+    private IConverter converter;
+
+    private static final String JSON = "application/json";
+
+    // This map stores all the information about the query
+    private Map<Integer, QueryInformation> queryStatusMap = new ConcurrentHashMap<>();
+
+    // Caches the query information for a day
+    private LoadingCache<Integer, Optional<QueryInformation>> queryStatusCache = CacheBuilder.newBuilder()
+            .maximumSize(1000).expireAfterWrite(1, TimeUnit.DAYS)
+            .removalListener(notification -> queryStatusMap.remove(notification.getKey()))
+            .build(new CacheLoader<Integer, Optional<QueryInformation>>() {
+                public Optional<QueryInformation> load(Integer key) throws Exception {
+                    return Optional.ofNullable(queryStatusMap.get(key));
+                }
+            });
+
+    /**
+     * {@inheritDoc}
+     * 
+     * @throws ExecutionException
+     */
+    @Override
+    public ExecutionStatus getQueryStatus(Integer queryID) throws ExecutionException {
+        Optional<QueryInformation> info = queryStatusCache.get(queryID);
+        return info.isPresent() ? info.get().getQueryStatus() : ExecutionStatus.UNKNOWN;
+    }
+
+    /**
+     * {@inheritDoc}
+     * 
+     * @throws ExecutionException
+     */
+    @Override
+    public void setQueryStatus(Integer queryID, ExecutionStatus status) throws ExecutionException {
+        Optional<QueryInformation> info = queryStatusCache.get(queryID);
+        if (info.isPresent()) {
+            info.get().setStatus(status);
+        } else {
+            queryStatusCache.put(queryID,
+                    Optional.ofNullable(new QueryInformation(status, queryID, new ArrayList<CreationEvent>())));
+        }
+
+    }
+
+    /**
+     * {@inheritDoc}
+     * 
+     * @throws ExecutionException
+     */
+    @Override
+    public void setQueryResult(Integer queryID, List<CreationEvent> res) throws ExecutionException {
+        Optional<QueryInformation> info = queryStatusCache.get(queryID);
+        info.get().setResult(res);
+    }
+
+    /**
+     * {@inheritDoc}
+     * 
+     * @throws ExecutionException
+     * @throws JSONException
+     */
+    @Override
+    public String getQueryResult(Integer queryID, String accept) throws ExecutionException, JSONException {
+        Optional<QueryInformation> info = queryStatusCache.get(queryID);
+
+        if (info.isPresent()) {
+            List<CreationEvent> elementList = info.get().getResult();
+            if (accept != null && accept.equals(JSON)) {
+                return converter.convertToJson(elementList);
+            }
+            return convertToCData(converter.convertToXML(elementList));
+        }
+        return "";
+    }
+
+    private String convertToCData(String xml) {
+        if (xml.isEmpty()) {
+            return xml;
+        }
+        return "<![CDATA[" + xml + "]]>";
+    }
```

Comment:
the same is still true: return proper xml here, no cdata. just add another method to the converter class that takes  an additional parameter (maybe list of additional values to be added to the root or whatever you need) and create proper XML.